### PR TITLE
Code refactoring to access blockchain services

### PR DESCRIPTION
**Pull request by lmahanand**
## Problem
Need to solve the intermittent issue that occurs when blockchain api service node tries to gossip with supervisor node.
Asana Link: 

## Solution

Implemented with Request and Reply apis provided by Noise P2P protocol. Now connections are closing gracefully for every request served.

## Testing Done and Results
Sanity testing.

Sent transactions through blockchain api service to blockchain using a client.
Retrieved Block Detail
Retrieved Account Detail

### CODE REVIEW

File: hbi/message/plugin.go [hunk 1]
```diff
@@ -111,11 +65,20 @@ func (state *TransactionMessagePlugin) Receive(ctx *network.PluginContext) error
 		tx := msg.GetTx()
 		accSrv := account.NewAccountService()
 		account, err := accSrv.GetAccountByAddress(msg.Tx.GetSenderAddress())
+		peerClient, err := ctx.Network().Client(ctx.Client().Address)
```
Comment (bittelc) [inline hbi/message/plugin.go]: What is a `peerClient`? I'm not sure if we can name this var any better, or if maybe I just need to understand something else.

Comment (bittelc) [reply hbi/message/plugin.go]: In this `ctx`, is `Client()` the API server?

File: hbi/message/plugin.go [hunk 2]
```diff
@@ -111,11 +65,20 @@ func (state *TransactionMessagePlugin) Receive(ctx *network.PluginContext) error
...
+		if err != nil {
+			return fmt.Errorf(fmt.Sprintf("Failed to get the peer client :%v", err))
```
Comment (bittelc) [inline hbi/message/plugin.go]: Again, what is a `peer client`?

File: hbi/message/plugin.go [hunk 3]
```diff
@@ -111,11 +65,20 @@ func (state *TransactionMessagePlugin) Receive(ctx *network.PluginContext) error
...
-			ctx.Network().BroadcastByAddresses(network.WithSignMessage(context.Background(), true),
+			err = peerClient.Reply(network.WithSignMessage(context.Background(), true), 1,
 				&protoplugin.TxResponse{
-					TxId: "", Status: "failed", Queued: 0, Pending: 0, Message: "Couldn't find the account due to : " + err.Error(),
-				}, ctx.Client().Address)
+					TxId: "", Status: "failed", Queued: 0, Pending: 0,
+					Message: "Couldn't find the account due to : " + err.Error(),
+				})
+			if err != nil {
+				return fmt.Errorf(fmt.Sprintf("Failed to reply to client :%v", err))
```
Comment (bittelc) [inline hbi/message/plugin.go]: typo, change to:
`client: %v`

File: hbi/message/plugin.go [hunk 4]
```diff
@@ -33,31 +33,12 @@ type BlockMessagePlugin struct{ *network.Plugin }
...
-				Address:     msg.GetAddress(),
-				Nonce:       account.Nonce,
-				Balance:     account.Balance,
-				StorageRoot: account.StorageRoot,
-				PublicKey:   account.PublicKey,
-			}
-			ctx.Network().BroadcastByAddresses(network.WithSignMessage(context.Background(), true), &accountResp, ctx.Client().Address)
-
-		}
+		findAccount(msg.Address, ctx)
```
[the remark this answers is not in the document]
Comment (lmahanand) [reply hbi/message/plugin.go]: done

File: hbi/message/plugin.go (hunk 1 above)
Comment (lmahanand) [reply hbi/message/plugin.go]: I have renamed **preerClient** to **apiClient**.  `ctx.Network().Client(ctx.Client().Address)` this line retrieves the connection detail of blockchain api server.

In this ctx, is Client() the API server? **Yes**

File: hbi/message/plugin.go (hunk 3 above)
Comment (lmahanand) [reply hbi/message/plugin.go]: done

File: hbi/message/plugin.go (hunk 2 above)
Comment (lmahanand) [reply hbi/message/plugin.go]: done and replied above

**Pull request merged**